libs/messenger-bundle/tests/ConnectionEvent/AuditLog/AuditEventFactoryTest.php

```diff
@@ -0,0 +1,97 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Keboola\MessengerBundle\Tests\ConnectionEvent\AuditLog;
+
+use Keboola\MessengerBundle\ConnectionEvent\AuditLog\AuditEventFactory;
+use Keboola\MessengerBundle\ConnectionEvent\AuditLog\GenericAuditLogEvent;
+use Keboola\MessengerBundle\ConnectionEvent\AuditLog\Project\ProjectDeletedEvent;
+use Keboola\MessengerBundle\ConnectionEvent\Exception\EventFactoryException;
+use PHPUnit\Framework\TestCase;
+
+class AuditEventFactoryTest extends TestCase
+{
+    public static function provideInvalidData(): iterable
+    {
+        yield 'no "data"' => [
+            'data' => [],
+            'error' => 'Missing or invalid property "data"',
+        ];
+
+        yield 'no "data.operation"' => [
+            'data' => [
+                'data' => [],
+            ],
+            'error' => 'Missing property "data.operation"',
+        ];
+    }
+
+    /** @dataProvider provideInvalidData */
+    public function testInvalidData(array $data, string $expectedError): void
+    {
+        $factory = new AuditEventFactory();
+
+        $this->expectException(EventFactoryException::class);
+        $this->expectExceptionMessage($expectedError);
+
+        $factory->createEventFromArray($data);
+    }
+
+    public function testErrorInDataMapping(): void
+    {
+        $factory = new AuditEventFactory();
+
+        $this->expectException(EventFactoryException::class);
+        $this->expectExceptionMessage(
+            'Failed to create Keboola\MessengerBundle\ConnectionEvent\AuditLog\Organization\ProjectCreatedEvent ' .
+            'from event data: Warning: Undefined array key "id"',
+        );
+
+        $factory->createEventFromArray([
+            'data' => [
+                'operation' => 'auditLog.organization.projectCreated',
+            ],
+        ]);
+    }
+
+    public function testCreateRegularEvent(): void
+    {
+        $factory = new AuditEventFactory();
+
+        $result = $factory->createEventFromArray([
+            'data' => [
+                'operation' => ProjectDeletedEvent::NAME,
```

Comment:
👍 ale měl bys to otestovat pro všechno z `AuditEventFactory::EVENTS` 🙏

Reply:
Tak jsme to udelal, jak jsme se bavili na stadupu - ke kazdemu eventu jsem pridal test, ze ho factory spravne vytvori (netestuju vsechny properties, jen ze vrati spravny typ eventu). Plus jsem to upravil tak, ze je oddeleny test na vytvoreni `fromArray`, export `toArray` a test na nevalidni `operation`.